### PR TITLE
Drop support for installing dependencies when resolving distutils commands

### LINKED ISSUE #3418: [BUG] performance after switching to nspektr / importlib_metadata as cratered
### setuptools version

setuptools>=60.8.0

### Python version

3.10.5

### OS

Arch Linux

### Additional environment information

_No response_

### Description

It's slow...

### Expected behavior

It's not slow.

### How to Reproduce

```shell
▸ python -m venv venv
▸ . ./venv/bin/activate
▸ python --version
Python 3.10.5
▸ pip install -U pip wheel setuptools
Requirement already satisfied: pip in ./venv/lib/python3.10/site-packages (22.0.4)
Collecting pip
  Using cached pip-22.1.2-py3-none-any.whl (2.1 MB)
Collecting wheel
  Using cached wheel-0.37.1-py2.py3-none-any.whl (35 kB)
Collecting setuptools
  Using cached setuptools-62.6.0-py3-none-any.whl (1.2 MB)
Installing collected packages: wheel, setuptools, pip
  Attempting uninstall: pip
    Found existing installation: pip 22.0.4
    Uninstalling pip-22.0.4:
      Successfully uninstalled pip-22.0.4
Successfully installed pip-22.1.2 setuptools-62.6.0 wheel-0.37.1
▸ wget -q https://files.pythonhosted.org/packages/71/39/171f1c67cd00715f190ba0b100d606d440a28c93c7714febeca8b79af85e/six-1.16.0.tar.gz
▸ tar xf six-1.16.0.tar.gz
▸ cd six-1.16.0
▸ time python setup.py -q bdist_wheel
python setup.py -q bdist_wheel  3.71s user 0.05s system 98% cpu 3.809 total
▸ pip install 'setuptools<60.10.0'
Collecting setuptools<60.10.0
  Downloading setuptools-60.9.3-py3-none-any.whl (1.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 17.1 MB/s eta 0:00:00
Installing collected packages: setuptools
  Attempting uninstall: setuptools
    Found existing installation: setuptools 62.6.0
    Uninstalling setuptools-62.6.0:
      Successfully uninstalled setuptools-62.6.0
Successfully installed setuptools-60.9.3
▸ time python setup.py -q bdist_wheel
python setup.py -q bdist_wheel  1.74s user 0.13s system 98% cpu 1.888 total
▸ pip install 'setuptools<60.8.0'
▸ time python setup.py -q bdist_wheel
python setup.py -q bdist_wheel  0.49s user 0.03s system 99% cpu 0.524 total
```

### Output

…

### PR DESCRIPTION
**Pull request by benoit-pierre**
## Summary of changes

Drop support for installing an entrypoint dependencies. This was added in `v0.6a0` (2005!), just before adding support for `setup_requires` in `v0.6a1`. The feature has been broken since `v60.8.0` (see https://github.com/pypa/setuptools/issues/3418#issuecomment-1167981467).

It's also unnecessary as it's the job of the code handling `setup_requires` to ensure all necessary build requirements are installed.

Note: `nspektr` has been unvendored as it's not needed anymore.

Closes #3418, partially supersede #3395.

### Pull Request Checklist
- [x] Changes have tests
- [x] News fragment added in [`changelog.d/`].

### CODE REVIEW

File: setuptools/tests/test_easy_install.py [hunk 1]
```diff
@@ -890,6 +890,72 @@ def test_setup_requires_with_transitive_extra_dependency(
                 monkeypatch.setenv(str('PIP_TIMEOUT'), str('0'))
                 run_setup(test_setup_py, [str('--version')])
 
+    def test_setup_requires_with_distutils_command_dep(self, monkeypatch):
```
Comment (jaraco) [inline setuptools/tests/test_easy_install.py:893]: Can we get a docstring to describe what behaviors this test is asserting?

File: setuptools/extern/__init__.py [hunk 2]
```diff
@@ -71,6 +71,6 @@ def install(self):
 
 names = (
     'packaging', 'pyparsing', 'ordered_set', 'more_itertools', 'importlib_metadata',
-    'zipp', 'importlib_resources', 'jaraco', 'typing_extensions', 'nspektr', 'tomli',
+    'zipp', 'importlib_resources', 'jaraco', 'typing_extensions', 'tomli',
```
Comment (jaraco) [inline setuptools/extern/__init__.py:74]: It's possible that removing nspektr also removes dependencies of nspektr. Let's confirm and remove them also if appropriate.

**Review by jaraco — APPROVED** [review]

File: setuptools/extern/__init__.py (hunk 2 above)
Comment (benoit-pierre) [reply setuptools/extern/__init__.py:74]: Will double check.

File: setuptools/tests/test_easy_install.py (hunk 1 above)
Comment (benoit-pierre) [reply setuptools/tests/test_easy_install.py:893]: Actually, I'm not sure there's a point in adding the test, as it's mostly testing the functionality being removed...

Comment (benoit-pierre) [reply setuptools/tests/test_easy_install.py:893]: There's only some partial overlap with `test_setup_requires_with_transitive_extra_dependency`, so better to keep it.

**Comment by benoit-pierre** [comment]
Ready for submission, @jaraco: do you want me to make a new release?

**Comment by jaraco** [comment]
Sure. Feel free to release anytime.

**Pull request merged**